pkg/build/api/v1/types.go

```diff
@@ -203,6 +207,25 @@ type ImageSourcePath struct {
 	DestinationDir string `json:"destinationDir" description:"relative destination directory in build home"`
 }
 
+// SecretBuildSource describes a secret and its destination directory that will be
+// used only at the build time. The content of the secret referenced here will
+// be copied into the destination directory instead of mounting.
+type SecretBuildSource struct {
+	// Secret is a reference to an existing secret that you want to use in your
+	// build.
+	Secret *kapi.LocalObjectReference `json:"secret,omitempty" description:"name of a secret to be used as a source"`
+
+	// DestinationDir is the directory where the files from the secret should be
+	// available for the build time.
+	// For the Source build strategy, these will be injected into a container
+	// where the assemble script runs. Later, when the script finishes, all files
+	// injected will be truncated to zero length.
+	// For the Docker build strategy, these will be copied into the build
```

Comment:
> can you elaborate on what you mean?

I don't see validation or restrictions preventing me from saying `../.../../foo`, but I can't see a good reason to allow that.